org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServerWrapper.java

```diff
@@ -314,6 +327,18 @@ private LanguageServerWrapper(@Nullable IProject project, LanguageServerDefiniti
 		final var errorsThreadNameFormat = formatPrefix + "#errorProcessor"; //$NON-NLS-1$
 		this.errorProcessor = Executors
 				.newSingleThreadExecutor(new ThreadFactoryBuilder().setNameFormat(errorsThreadNameFormat).build());
+
+		Path watchedFilesBasePath = null;
+		try {
+			if (initialProject != null) {
+				final var loc = initialProject.getLocationURI();
+				if (loc != null)
+					watchedFilesBasePath = Paths.get(loc);
+			}
+		} catch (IllegalArgumentException ex) {
+			LanguageServerPlugin.logError(ex);
+		}
+		this.fileSystemWatcherManager = new FileSystemWatcherManager(watchedFilesBasePath);
```

Comment:
To keep the LSW simple, would it make sense to just pass `IProject` to its constructor and move all the preceding code into that constructor?